src/device-registry/models/Event.js

```diff
@@ -355,45 +357,105 @@ eventSchema.statics = {
     delete search["metadata"];
     delete search["tenant"];
 
+    if (!metadata || metadata === "device" || metadata === "device_id") {
+      groupId = "$" + metadata ? metadata : groupId;
+      localField = metadata ? metadata : localField;
+      if (metadata === "device_id") {
+        foreignField = "_id";
+      }
+      if (metadata === "device" || !metadata) {
+        foreignField = "name";
+      }
+
+      from = "devices";
+      _as = "_deviceDetails";
+      as = "deviceDetails";
+      elementAtIndex0 = { $arrayElemAt: ["$deviceDetails", 0] };
+
+      projection[as] = {};
```

Comment:
Please note that there is code duplication in this PR, there is no good application of "code reuse". Delighted to inform you that this issue is being handled by another ongoing [refactoring (pagination) PR](https://github.com/airqo-platform/AirQo-api/pull/674). @ja-odur @NoahNsimbe